### PR TITLE
fix: add hide PR button under pipeline options with remote support

### PR DESCRIPTION
**Pull request by adong**
## Context

Follow up of https://github.com/screwdriver-cd/ui/pull/665
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/workflow-graph-d3/component.js [hunk 1]
```diff
@@ -142,6 +154,7 @@ export default Component.extend({
 
     let displayJobNameLength = ENV.APP.MINIMUM_JOBNAME_LENGTH;
 
+    // const pipelinePreference = await this.shuttle.getUserPreference(this.get('pipeline.id'));
```
Comment (tkyi) [inline app/components/workflow-graph-d3/component.js:157]: commented code

File: app/components/pipeline-options/component.js [hunk 2]
```diff
@@ -216,6 +216,31 @@ export default Component.extend({
...
         this.set('isUpdatingMetricsDowntimeJobs', false);
       }
+    },
+
+    async updateShowPRJobs(showPRJobs) {
+      const pipelineId = this.get('pipeline.id');
+
+      let pipelinePreference = await this.store.queryRecord('preference/pipeline', {
+        filter: { pipelineId }
+      });
```
Comment (tkyi) [inline app/components/pipeline-options/component.js:226]: Should you just use the `getUserPreference` method here?
```suggestion
      let pipelinePreference = await this.shuttle.getUserPreference(pipelineId);
```

File: app/components/pipeline-options/component.js [hunk 3]
```diff
@@ -216,6 +216,31 @@ export default Component.extend({
...
+
+      if (pipelinePreference) {
+        pipelinePreference.set('showPRJobs', showPRJobs);
```
Comment (tkyi) [inline app/components/pipeline-options/component.js:229]: any reason why you use `pipelinePreference.set` here vs `pipelinePreference = x` below?

File: app/shuttle/service.js [hunk 4]
```diff
@@ -185,5 +186,115 @@ export default Service.extend({
     const url = `/pipelines/${pipelineId}/latestCommitEvent`;
 
     return this.fetchFromApi(method, url);
+  },
+
+  /**
+   * get
```
Comment (tkyi) [inline app/shuttle/service.js:192]:
```suggestion
   * getUserSetting
```

File: app/shuttle/service.js [hunk 5]
```diff
@@ -185,5 +186,115 @@ export default Service.extend({
...
+  },
+
+  /**
+   * updateUserPreference
+   * @param  {Number}  [pipelineId]
+   * @param  {Object}  pipelineSettings
+   * @param  {Boolean} [pipelineSettings.showPRJobs]
+   * @return {Promise}
+   */
+  async updateUserPreference(pipelineId, pipelineSettings) {
```
Comment (tkyi) [inline app/shuttle/service.js:287]: what's the difference between this method and `updateUserSetting`?

Comment (adong) [reply app/shuttle/service.js:287]: They are the same, `updateUserSetting ` is now removed. I was doing some refactoring for `updateUserSetting` and `getUserSetting` back then, and didn't remove.

File: app/components/pipeline-options/component.js (hunk 3 above)
Comment (adong) [reply app/components/pipeline-options/component.js:229]: Could be either way. Updated.

File: app/components/pipeline-options/component.js (hunk 2 above)
Comment (adong) [reply app/components/pipeline-options/component.js:226]: `getUserPreference` will trigger an API call to compare local and remove settings.

```const remotePreferences = await this.getUserSetting();```

**Review by tkyi — APPROVED** [review]

**Pull request merged**